File: backend/src/categories/categories.service.ts
import { Injectable } from '@nestjs/common';
import { SupabaseService } from '../supabase/supabase.service';

@Injectable()
export class CategoriesService {
  constructor(private readonly supabaseService: SupabaseService) {}

  async getAllCategories() {
    const { data, error } = await this.supabaseService.client
      .from('categories')
      .select('*')
      .order('id', { ascending: true });

    if (error) {
      throw new Error(`Failed to fetch categories: ${error.message}`);
    }

    return data;
  }
}

File: backend/app/services/categories_service.py
from app.services.supabase_service import supabase_service


def get_all_categories() -> list[dict]:
    response = (
        supabase_service.client.table("categories")
        .select("*")
        .order("id", desc=False)
        .execute()
    )

    return response.data or []
